modules/ui/rapid_view_manage_datasets.js

```diff
@@ -216,30 +228,88 @@ export function uiRapidViewManageDatasets(context, parentModal) {
 
 
   function renderDatasets(selection) {
+    const status = selection.selectAll('.rapid-view-manage-datasets-status');
+    const results = selection.selectAll('.rapid-view-manage-datasets');
+
     const showPreview = prefs('rapid-internal-feature.previewDatasets') === 'true';
     const service = services.esriData;
+
     if (!service || (Array.isArray(_datasetInfo) && !_datasetInfo.length)) {
-      selection.text(t('rapid_feature_toggle.esri.no_datasets'));
+      results.classed('hide', true);
+      status.classed('hide', false).text(t('rapid_feature_toggle.esri.no_datasets'));
       return;
     }
 
     if (!_datasetInfo) {
-      selection.text(t('rapid_feature_toggle.esri.fetching_datasets'));
+      results.classed('hide', true);
+      status.classed('hide', false)
+        .text(t('rapid_feature_toggle.esri.fetching_datasets'));
+
+      status
+        .append('br');
+
+      status
+        .append('img')
+        .attr('class', 'rapid-view-manage-datasets-spinner')
+        .attr('src', context.imagePath('loader-black.gif'));
+
       service.loadDatasets()
         .then(results => {
-          // exclude preview datasets unless user has opted into them
-          return _datasetInfo = Object.values(results)
-            .filter(d => showPreview || !d.groupCategories.some(category => category === '/Categories/Preview'));
+          // Build set of available categories
+          let categories = new Set();
+
+          Object.values(results).forEach(d => {
+            d.groupCategories.forEach(c => {
+              categories.add(c.toLowerCase().replace('/categories/', ''));
+            });
+          });
+          if (!showPreview) categories.delete('preview');
+
+          const combodata = Array.from(categories).sort().map(c => {
+            let item = { title: c, value: c };
+            if (c === 'preview') item.display = `${c} <span class="rapid-view-manage-dataset-beta beta"></span>`;
+            return item;
+          });
+          categoryCombo.data(combodata);
+
+          // Exclude preview datasets unless user has opted into them
+          _datasetInfo = Object.values(results)
+            .filter(d => showPreview || !d.groupCategories.some(category => category.toLowerCase() === '/categories/preview'));
+
+          return _datasetInfo;
         })
         .then(() => _content.call(renderModalContent));
+
       return;
     }
 
-    selection.text('');
+    results.classed('hide', false);
+    status.classed('hide', true);
+
+    // Apply filters
+    let count = 0;
+    _datasetInfo.forEach(d => {
+      count++;
+      const title = (d.title || '').toLowerCase();
```

Comment:
I see you have done case-insensitive compares... I see this is NOT your first text filtering rodeo. Well played. I will get you next time...